docs/AUTHENTICATION.md

```diff
@@ -178,3 +178,168 @@ EOF
 
 Now that you have configured `provider-gcp` with Workload Identity supported,
 you can [provision infrastructure](https://crossplane.io/docs/v1.6/getting-started/provision-infrastructure).
+
+
+## Authenticating with Access Tokens
+
+Using temporary Access Tokens will require a process to regenerate an access token before it expires. Luckily we can use a Kubernetes CronJob to fulfill that.
+
+### Steps
+
+#### 0. Prepare your variables
+
+In the following sections, you'll need to name your resources.
+Define the variables below with any names valid in Kubernetes or GCP so that you
+can smoothly set it up:
+
+```console
+$ PROJECT_ID=<YOUR_GCP_PROJECT_ID>                               # e.g.) acme-prod
+$ GCP_SERVICE_ACCOUNT=<YOUR_CROSSPLANE_GCP_SERVICE_ACCOUNT_NAME> # e.g.) crossplane
+$ ROLE=<YOUR_ROLE_FOR_CROSSPLANE_GCP_SERVICE_ACCOUNT>            # e.g.) roles/cloudsql.admin
+$ KUBERNETES_SERVICE_ACCOUNT=<YOUR_KUBERNETES_SERVICE_ACCOUNT>   # e.g.) token-generator
+$ NAMESPACE=<YOUR_KUBERNETES_NAMESPACE>                          # e.g.) mynamespace
+$ SECRET_NAME=<YOUR_CREDENTIALS_SECRET_NAME>                     # e.g.) gcp-credentials
+$ SECRET_KEY=<NAME_OF_KEY_IN_SECRET>                             # e.g.) token
+```
+
+#### 1. Configure service accounts to use Workload Identity
+
+Create a GCP service account, which will be used for provisioning actual
+infrastructure in GCP, and grant IAM roles you need for accessing the Google
+Cloud APIs:
+
+```console
+$ gcloud iam service-accounts create ${GCP_SERVICE_ACCOUNT} --project ${PROJECT_ID}
+$ gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com" \
+    --role ${ROLE} \
+    --project ${PROJECT_ID}
+```
+
+#### 2. Create resources to generate an access-token
+Create the Kubernetes service account, RBAC, and CronJob to generate the temporary access-token
+```console
+$ cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name:  ${KUBERNETES_SERVICE_ACCOUNT}
+  namespace: ${NAMESPACE}
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
+rules:
+- apiGroups: [""]
+  resources:
+  - secrets
+  verbs:
+  - get
+  - create
+  - patch
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync-rb
+subjects:
+- kind: ServiceAccount
+  name: ${KUBERNETES_SERVICE_ACCOUNT}
+roleRef:
+  kind: Role
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
+  apiGroup: ""
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}-credentials-sync
+spec:
```

Comment:
I think you need to add in the `metadata` object of these resources the `namespace: ${NAMESPACE}` line. If we try to run the tutorial as it is, the job created by the command `kubectl create job` will fail. This was the error described in the Events:

```console
Events:
  Type     Reason        Age                  From            Message
  ----     ------        ----                 ----            -------
  Warning  FailedCreate  25s (x6 over 5m35s)  job-controller  Error creating: pods "cred-sync-001--1-" is forbidden: error looking up service account default/token-generator: serviceaccount "token-generator" not found
```

Reply:
I ran through the Auth example again and didn't hit this issue. Did you happen to set the NAMESPACE environment variable?

Reply:
Yes, I have. Did you set the NAMESPACE environment variable to `default`? If so, that's why it's working for you.

If you set the NAMESPACE to a value other than `default`, it won't work. That's because in the docs you are creating the `ServiceAccount` in the namespace of the variable `NAMESPACE`. See below:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name:  ${KUBERNETES_SERVICE_ACCOUNT}
  namespace: ${NAMESPACE} # Here you are setting the namespace
```

Then you are creating the Role and RoleBinding in the `default` namespace:

```yaml
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
  # Missing namespace field, Role will be created in the default namespace
rules:
- apiGroups: [""]
  resources:
  - secrets
  verbs:
  - get
  - create
  - patch
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync-rb
  # Missing namespace field, RoleBinding will be created in the default namespace
subjects:
- kind: ServiceAccount
  name: ${KUBERNETES_SERVICE_ACCOUNT} # This ServiceAccount doesn't exists in the default namespace
roleRef:
  kind: Role
  name: ${KUBERNETES_SERVICE_ACCOUNT}-sync
  apiGroup: ""
```

The problem here is that the `subject` attribute of the RoleBinding is referencing a ServiceAccount that doesn't exist (because this ServiceAccount doesn't exists in the `default` namespace). You need to add the `namespace` attribute to the `metadata` object of the Role, RoleBinding and CronJob manifests. They are all using the `default` namespace.